Templating/Helper/DateTimeHelper.php

```diff
@@ -140,12 +142,14 @@ public function format($datetime, $pattern, $locale = null, $timezone = null)
     }
 
     /**
+     * NEXT_MAJOR: Change to $date to \DateTimeInterface.
+     *
      * @param \IntlDateFormatter $formatter
-     * @param \Datetime          $date
+     * @param \DateTime          $date
      *
      * @return string
      */
-    public function process(\IntlDateFormatter $formatter, \Datetime $date)
+    public function process(\IntlDateFormatter $formatter, \DateTime $date)
```

Comment:
Yes and no, `process()` will not yet support `DateTimeImmutable`. However, all other methods use `getDatetime()` before passing the result to `process()`. Therefore we can use `DateTimeImmutable` in all templating helpers and Twig filters.